objects/src/accounts/package/storage_entry/word.rs

```diff
@@ -0,0 +1,266 @@
+    Dynamic(TemplateKey),
+}
+
+impl FeltRepresentation {
+    pub fn template_keys(&self) -> impl Iterator<Item = &TemplateKey> {
+        let maybe_key = match self {
+            FeltRepresentation::Dynamic(template_key) => Some(template_key),
+            _ => None,
+        };
+
+        maybe_key.into_iter()
+    }
+
+    /// Attempts to convert the [FeltRepresentation] into a [Felt].
+    ///
+    /// If the representation is dynamic, the value is retrieved from `template_values`, identified
+    /// by its key. Otherwise, the returned value is just the inner element.
+    pub fn try_into_felt(
+        self,
+        template_values: &BTreeMap<String, TemplateValue>,
+    ) -> Result<Felt, AccountComponentTemplateError> {
+        match self {
+            FeltRepresentation::SingleHex(base_element) => Ok(base_element),
+            FeltRepresentation::SingleDecimal(base_element) => Ok(base_element),
+            FeltRepresentation::Dynamic(template_key) => template_values
+                .get(template_key.inner())
+                .ok_or_else(|| {
+                    AccountComponentTemplateError::TemplateValueNotProvided(
+                        template_key.inner().to_string(),
+                    )
+                })?
+                .as_felt()
+                .copied(),
+        }
+    }
+}
+
+impl Default for FeltRepresentation {
+    fn default() -> Self {
+        FeltRepresentation::SingleHex(Felt::default())
+    }
+}
+
+impl<'de> Deserialize<'de> for FeltRepresentation {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value = String::deserialize(deserializer)?;
+        if let Some(hex_str) = value.strip_prefix("0x").or_else(|| value.strip_prefix("0X")) {
+            let felt_value = u64::from_str_radix(hex_str, 16).map_err(serde::de::Error::custom)?;
+            Ok(FeltRepresentation::SingleHex(Felt::new(felt_value)))
+        } else if let Ok(decimal_value) = value.parse::<u64>() {
+            Ok(FeltRepresentation::SingleDecimal(Felt::new(decimal_value)))
+        } else if let Ok(key) = TemplateKey::try_from(&value) {
+            Ok(FeltRepresentation::Dynamic(key))
+        } else {
+            Err(serde::de::Error::custom("Value is not a valid element"))
+        }
+    }
+}
+
+impl Serialize for FeltRepresentation {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            FeltRepresentation::SingleHex(felt) => {
+                let output = format!("0x{:x}", felt.as_int());
+                serializer.serialize_str(&output)
+            },
+            FeltRepresentation::SingleDecimal(felt) => {
+                let output = felt.as_int().to_string();
+                serializer.serialize_str(&output)
+            },
+            FeltRepresentation::Dynamic(key) => key.serialize(serializer),
+        }
+    }
+}
```

Comment:
Should we add some roundtrip tests for serialization for `FeltRepresentation` and `WordRepresentation`?

Reply:
> I'm also wondering if making everything pub by default is good. There are a lot of new types added in this PR and the API surface of miden-base increases quite a bit. If we only export some of the higher level types, it would be great to double-check if the lower level types are properly excluded from the docs and that they are not exposed through methods on the higher level types (although rustdoc should complain in that case anyway).

Yeah, I left a `TODO` to address this because I wanted to double check what we would need and restrict everything from there, but definitely agree that it should not all be `pub` by default and was planning to change this.

Reply:
I tried this before, but it's not possible:

https://docs.rs/toml/0.7.2/toml/ser/struct.Serializer.html
> Note that the TOML format does not support all datatypes in Rust, such as enums, tuples, and tuple structs. These types will generate an error when serialized.

You get an error like this one:

```
called `Result::unwrap()` on an `Err` value: Error { inner: UnsupportedType(None) }
```

It is tested through the use of the full format